src/main/java/seedu/address/model/attendance/AttendancePresence.java

```diff
@@ -0,0 +1,72 @@
+package seedu.address.model.attendance;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the presence of a student in an attendance record.
+ * It can either be 'present' or 'absent'.
+ */
+public class AttendancePresence {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Presence field should only contain either 'present' or 'absent', and it should not be blank";
+    public static final String VALID_REGEX = "(?i)^(present|absent)$";
```

Comment:
Not sure if it will be better, but have you considered using enum?

Reply:
Great idea! Refactored AttendancePresence to use enum